src/encoded/static/components/search.js

```diff
@@ -786,11 +787,18 @@ class Facet extends React.Component {
         // We have to check the full list for now (until schema change) because some lists contain both numerical and string terms ('Encyclopedia version' under Annotations) and we do not want to sort those by value
         const numericalTest = a => !isNaN(a.key);
         // For date facets, sort by date
-        if (field.match('date') || field.match('month') || field.match('year')) {
-            terms.sort((a, b) => (new Date(b.key) - new Date(a.key)));
+        let terms = [];
+        if (field.match('date')) {
+            terms = _.sortBy(unsortedTerms, obj => moment(obj.key, 'YYYY-MM-DD').utc());
+        } else if (field.match('month')) {
+            terms = _.sortBy(unsortedTerms, obj => moment(obj.key, 'MMMM, YYYY').utc());
+        } else if (field.match('year')) {
+            terms = _.sortBy(unsortedTerms, obj => moment(obj.key, 'YYYY').utc());
         // For straightforward numerical facets, just sort by value
-        } else if (terms.every(numericalTest)) {
-            terms.sort((a, b) => (a.key - b.key));
+        } else if (unsortedTerms.every(numericalTest)) {
+            terms = _.sortBy(unsortedTerms, obj => +obj.key);
```

Comment:
What’s the “+” in the iteratee for?

Reply:
That was just to confirm that obj.key was a number not a string.

Reply:
(It is a number so it's not necessary, it's just a habit because sometimes JS interprets numbers as strings and then there can be very unexpected behavior.)

Reply:
That sounds vaguely familiar now, and it’s actually ECMA sanctioned: http://www.ecma-international.org/ecma-262/5.1/#sec-11.4.7

Reply:
Oh whoops, I didn't see this till now. Should I add it back?

Reply:
Oh, it’s up to you. I just didn’t know about it.